hazelcast/src/main/java/com/hazelcast/spi/impl/merge/MapMergingEntryImpl.java

```diff
@@ -210,15 +207,11 @@ public void writeData(ObjectDataOutput out) throws IOException {
         out.writeLong(lastUpdateTime);
         out.writeLong(version);
         out.writeLong(ttl);
-        // RU_COMPAT_3_10
         // WAN events received from source cluster also carry null maxIdle
-        // even when cluster version is 3.11
-        if (out.getVersion().isGreaterOrEqual(V3_11)) {
-            boolean hasMaxIdle = maxIdle != null;
-            out.writeBoolean(hasMaxIdle);
-            if (hasMaxIdle) {
-                out.writeLong(maxIdle);
-            }
+        boolean hasMaxIdle = maxIdle != null;
```

Comment:
`maxIdle` field has comment over: `RU_COMPAT_3_10 (Long -> long)`. Does that mean this field should be converted to `long`?

Reply:
It looked a bit tricky so I wanted to leave it for a separate PR as there are many places where this was affecting. I'll do a deeper dive then instead.

Reply:
Added comments for clarification. 3.10 members didn't send `maxIdle` (does not apply anymore) and entries received via WAN don't send `maxIdle` (still applies, potentially we can send this information as well, I'll think about it).